pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/ServerCnx.java

```diff
@@ -361,8 +362,9 @@ protected void handleProducer(final CommandProducer cmdProducer) {
                         // until the previous producer creation
                         // request
                         // either complete or fails.
+                        ServerError error = !existingProducerFuture.isDone() ? ServerError.ServiceNotReady : ServerError.UnknownError;
                         log.warn("[{}][{}] Producer is already present on the connection", remoteAddress, topicName);
-                        ctx.writeAndFlush(Commands.newError(requestId, ServerError.UnknownError,
+                        ctx.writeAndFlush(Commands.newError(requestId, error,
```

Comment:
This is:  producer-creation is still in process usecase. therefore, returning ```ServiceNotReady``` is a retriable error at client side and it can be reconnected later on.